### PR TITLE
Fix: floating point literals must use the f suffix when working with float variables

### PR DESCRIPTION
**Pull request by obones**
This fixes C4244/C4267/C4305 warnings related to `double` to `float` conversions when a literal is involved in the warning, as reported in #1866

### CODE REVIEW

**Comment by merbanan** [comment]
LGTM, but I'll wait abit before committing.

File: src/devices/acurite.c [hunk 1]
```diff
@@ -729,7 +729,7 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
             // 11 bits needed for specified range -40 C to 70 C (-40 F - 158 F)
             // range -100 C to 1538.4 C
             int temp_raw = ((bb[4] & 0x7F) << 7) | (bb[5] & 0x7F);
-            tempc = temp_raw * 0.1 - 100;
+            tempc = temp_raw * 0.1f - 100;
```
Comment (zuckschwerdt) [inline src/devices/acurite.c:732]: there is still a double lurking here. prefered would be `(temp_raw - 1000) * 0.1f`

Comment (zuckschwerdt) [reply src/devices/acurite.c:732]: or maybe it's an implicit float…

Comment (obones) [reply src/devices/acurite.c:732]: According to VC++, it gets indeed converted to a `float` not a `double`, there is no longer a warning on that line.

Comment (obones) [reply src/devices/acurite.c:732]: Note that I hesitated changing `100` to `100.0f` but I'm not sure it makes it any clearer...

Comment (merbanan) [reply src/devices/acurite.c:732]: temp_raw should be converted to float and then -100 should be converted to float and then everything is float

Comment (zuckschwerdt) [reply src/devices/acurite.c:732]: my point was an integer substraction, then a float multiply instead of two float ops.

Comment (merbanan) [reply src/devices/acurite.c:732]: The rules give that it should be float. But I see your point if the int is large.

Comment (obones) [reply src/devices/acurite.c:732]: The initial goal of that PR was to address the warnings by simply adding the suffix wherever required. 
That being said, I agree that there are optimization opportunities at various places (`value / 3600.0f / 1000.0f` comes to mind) but shouldn't this be addressed in another PR?

Comment (zuckschwerdt) [reply src/devices/acurite.c:732]: True, and greping for `f - ` afterwards will find most of those instances.

File: src/devices/acurite.c [hunk 2]
```diff
@@ -588,7 +588,7 @@ static int acurite_atlas_decode(r_device *decoder, bitbuffer_t *bitbuffer, unsig
         /* clang-format off */
         data = data_append(data,
                 "wind_dir_deg",     NULL,           DATA_FORMAT,    "%.1f",         DATA_DOUBLE, wind_dir,
-                "rain_in",          "Rainfall Accumulation", DATA_FORMAT, "%.2f in", DATA_DOUBLE, raincounter * 0.01f,
+                "rain_in",          "Rainfall Accumulation", DATA_FORMAT, "%.2f in", DATA_DOUBLE, raincounter * 0.01,
```
Comment (merbanan) [inline src/devices/acurite.c:591]: In this case I actually wanted it to be a float but it seems to be missed from the C-standard. Or something like that.

**Comment by obones** [comment]
I have just rebased this branch on the latest master, please let me know if this is still of interest

File: src/devices/acurite.c (hunk 2 above)
Comment (gdt) [reply src/devices/acurite.c:921]: I'm going to interpret this as 1) you would have liked to have DATA_FLOAT as a format and used it  2) given that it says DOUBLE, this change is ok.   Of course speak up if that second point is a  misinterpretation; the first point doesn't really bear on "should we merge this".

**Comment by gdt** [comment]
This looks ready to merge to me, by code review.

**Pull request merged**